### PR TITLE
Escape values to be displayed in token labels

### PR DESCRIPTION
**Pull request by DanThiffault**
If the value of a jquery.token source contained either { or } the
highlight regex would fail with an exception as documented here
 https://github.com/loopj/jquery-tokeninput/issues/785

Since the upstream project seems to be abandoned, making the fix in our
vendored version of the library. Also updating to the most recent verion
of the active admin 1.0.x release candidate.

It seems like the specs in this project have been broken for a while,
including the updated dummy db schema as a step towards getting those
straightened out.

### CODE REVIEW

File: vendor/assets/javascripts/jquery.tokeninput.js [hunk 1]
```diff
@@ -699,11 +699,16 @@ $.TokenList = function (input, url_or_data, settings) {
 
     // Highlight the query part of the search term
     function highlight_term(value, term) {
-        return value.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + term + ")(?![^<>]*>)(?![^&;]+;)", "gi"), "<b>$1</b>");
+        return value.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + escape_regexp_chars(term) + ")(?![^<>]*>)(?![^&;]+;)", "gi"), "<b>$1</b>");
```
Comment (thegreatape) [inline vendor/assets/javascripts/jquery.tokeninput.js:702]: look_away_marian.gif

File: vendor/assets/javascripts/jquery.tokeninput.js [hunk 2]
```diff
@@ -699,11 +699,16 @@ $.TokenList = function (input, url_or_data, settings) {
...
     }
 
     function find_value_and_highlight_term(template, value, term) {
-        return template.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + value + ")(?![^<>]*>)(?![^&;]+;)", "g"), highlight_term(value, term));
+        return template.replace(new RegExp("(?![^&;]+;)(?!<[^<>]*)(" + escape_regexp_chars(value) + ")(?![^<>]*>)(?![^&;]+;)", "g"), highlight_term(value, term));
+    }
+
+    function escape_regexp_chars(string) {
+        var specials = new RegExp("[.*+?|()\\[\\]{}\\\\]", "g"); // .*+?|()[]{}\
```
Comment (thegreatape) [inline vendor/assets/javascripts/jquery.tokeninput.js:710]: I'm assuming the comment here is the set of characters escaped?

**Comment by thegreatape** [comment]
Looks good, despite eyeball-bleeding regexes.

File: vendor/assets/javascripts/jquery.tokeninput.js (hunk 2 above)
Comment (DanThiffault) [inline vendor/assets/javascripts/jquery.tokeninput.js:710]: That's the intention. I've tested at least half of them.

**Pull request merged**